upgrade-job/src/common/clients.rs

```diff
@@ -0,0 +1,56 @@
+use crate::common::error::{
+    Error::{K8sClientGeneration, RestClientConfiguration, RestUrlParse},
+    Result,
+};
+use kube::Client;
+use openapi::tower::client::{ApiClient, Configuration as rest_config};
+use std::time::Duration;
+use tokio::sync::OnceCell;
+use url::Url;
+
+/// Thread-safe global storage REST API client container.
+static REST_CLIENT: OnceCell<ApiClient> = OnceCell::const_new(); // Use rest_client().
+
+pub(crate) async fn get_or_init_rest_client(rest_endpoint: &str) -> Result<&ApiClient> {
```

Comment:
if you have objects consider keeping the clients within them rather than a global, if possible.

Reply:
Yes, I've removed global clients.